### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by joaobrittoneto**
@doudou 
Made some debug and tests in the tank. 
So far so good. 
How can I run a operation in rock-display?

### CODE REVIEW

File: tasks/Task.cpp [hunk 1]
```diff
@@ -87,141 +208,34 @@ void Task::stopHook()
...
     TaskBase::cleanupHook();
+
+    //Make sure to set the device to its default operational mode, DATA.
+    if(driver->getMode() == COMMAND)
+        driver->switchToDataMode();
+
 }
 void Task::processIO()
 {
-    AcousticConnection acoustic_connection = driver->getConnectionStatus();
```
Comment (doudou) [inline tasks/Task.cpp:93]: You should definitely put the reading bits in processIO()

File: tasks/Task.cpp [hunk 2]
```diff
@@ -229,7 +243,21 @@ void Task::checkFreeBuffer(std::string const &buffer, AcousticConnection const &
...
+// Filter possible <+++ATcommand> in raw_data_input
+void Task::filterRawData( std::string const & raw_data_in)
+{
+    // Find "+++"
+    if (raw_data_in.find("+++") != string::npos)
+    {
+        std::string error_msg = "Usbl_evologics Task.cpp. There is the malicious string \"+++\" in raw_data_input. DO NOT use it as it could be interpreted as a command by device.";
+        std::cout << error_msg << std::endl;
+        RTT::log(RTT::Error) << error_msg << std::endl;
+        exception(MALICIOUS_SEQUENCE_IN_RAW_DATA);
```
Comment (doudou) [inline tasks/Task.cpp:284]: exception() does not throw (it's only a state change), so you will still send the bad message. You could throw just after having called exception().

File: tasks/Task.cpp [hunk 3]
```diff
@@ -229,38 +258,81 @@ void Task::checkFreeBuffer(std::string const &buffer, AcousticConnection const &
...
+        throw;
```
Comment (doudou) [inline tasks/Task.cpp]: throw without arguments is only valid in a catch() context. You have to specify an exception object (e.g. std::runtime_error("error message"))

**Comment by joaobrittoneto** [comment]
@doudou 
Better understanding of instant message retries in case of delivery failure.
Remove cout's.
Include throw with runtime_error .
Log current device's settings.

**Pull request merged**